### PR TITLE
Backport 2.7: update test RSA certificate to use SHA-256 instead of SHA-1

### PR DESCRIPTION
**Pull request by RonEld**
## Description
Backport of #1520 to `mbedtls-2.7`

## Status
**READY**

### CODE REVIEW

File: library/certs.c [hunk 1]
```diff
@@ -173,14 +170,72 @@ static const char mbedtls_test_ca_crt_rsa_sha256[] = TEST_CA_CRT_RSA_SHA256;
...
 
-#endif
+const char mbedtls_test_srv_crt_rsa[]     =  TEST_SRV_CRT_RSA_SHA256;
+const size_t mbedtls_test_srv_crt_rsa_len = sizeof( mbedtls_test_srv_crt_rsa );
+#define TEST_SRV_CRT_RSA_SOME
+#endif /* MBEDTLS_SHA256_C */
+
+#if !defined(TEST_SRV_CRT_RSA_SOME) || defined(MBEDTLS_SHA1_C)
+/* tests/data_files/server2.crt */
+#define TEST_SRV_CRT_RSA_SHA1                                                  \
```
Comment (AndrzejKurek) [inline library/certs.c]: The indentation of "\" is off in this PR.

File: library/certs.c [hunk 2]
```diff
@@ -173,14 +170,72 @@ static const char mbedtls_test_ca_crt_rsa_sha256[] = TEST_CA_CRT_RSA_SHA256;
...
+"-----BEGIN CERTIFICATE-----\r\n"                                              \
+"MIIDNzCCAh+gAwIBAgIBAjANBgkqhkiG9w0BAQUFADA7MQswCQYDVQQGEwJOTDER\r\n"         \
```
Comment (AndrzejKurek) [inline library/certs.c]: Why does this certificate differ, compared to 2.16 and the original PR?

Comment (RonEld) [reply library/certs.c]: the change is from [this commit](https://github.com/ARMmbed/mbedtls/commit/beffcd8d4acfd8a01ecf23a3a622a91ebb81e732)
You can see that the hardcoded certificate is same as the original certificate in this branch

File: library/certs.c (hunk 1 above)
Comment (RonEld) [reply library/certs.c]: Please elaborate. where should it be? I aligned all the `\` chars.

**Comment by RonEld** [comment]
@AndrzejKurek Thank you for the review.
I answered your questions.

File: library/certs.c (hunk 2 above)
Comment (RonEld) [reply library/certs.c]: Note: The commit that updated the hard-coded certificates was not backported to this PR.
In addition, this data will be updated with backports of #2417

File: library/certs.c (hunk 1 above)
Comment (AndrzejKurek) [reply library/certs.c]: I meant aligning it the same way as it is done in the other backport (or the original PR), or at lines 184-202.

File: library/certs.c (hunk 2 above)
Comment (AndrzejKurek) [reply library/certs.c]: OK, thanks.

**Comment by RonEld** [comment]
@AndrzejKurek I fixed the alignment of `\` character, per your comment.
Since it is a small enough change, I amended to previous commit.

File: library/certs.c (hunk 1 above)
Comment (RonEld) [reply library/certs.c]: note I have just noticed that the alignment in SHA1 certificate isnot the same as in the SHA256 certificate, for the other backports as well. I don't think this should block merging though, I will fix this in #2417 and its backports

**Comment by RonEld** [comment]
ping @AndrzejKurek @k-stachowiak

**Pull request merged**